### ISSUE TITLE
Simultaneous Parameter Estimation

### ISSUE THREAD

**Issue by Theo-BRN** [copasi/basico#50]
Heya, hope you're well!

I was wondering if simultaneous parameter estimation was possible with copasi/basico. For example, if you had two experiments you want to fit to, and of the two parameters to fit (k1 and k2), each experiment has a different k1, but a shared k2. Can you essentially tell copasi "this parameter is the same in both experiments, please fit it" and "this parameter may be different in each experiment, please fit it".

I know you can declare what experiments are affected by a parameter with `set_fit_parameters` but I'm not sure that's quite the same. 

Any clarification, or reference to examples in documentation would be greatly appreciated.

Thanks,
Theo

**Comment by fbergmann** [comment]
You are completely right, that is what affected experiments are made for. If not specified the parameter will be assumed to be the *same* for *all* experiments. Otherwise that parameter would be only used for the experiments specified. So if you had a different k1, you'd have 2 fit items for k1, with different affected experiments. I hope this makes it clear.

**Comment by Theo-BRN** [comment]
Ah great, thank you for the clarification, that does help!

Is it possible to have a parameter be fitted as shared for some experiments and then independent for the rest? Let's say experiments 1, 2 and 3 share the same k_on variable and experiments 4 and 5 have different k_on variables. Perhaps the `fit_parameters` argument would look like:

```
fit_items = [
    {'name': "Values[k_on]", 'lower': 1e05, 'upper': 1e09, 'affected': ['Exp_1, Exp_2, Exp_3']},
    {'name': "Values[k_on]", 'lower': 1e05, 'upper': 1e09, 'affected': ['Exp_4']},
    {'name': "Values[k_on]", 'lower': 1e05, 'upper': 1e09, 'affected': ['Exp_5']},
]

```

**Comment by fbergmann** [comment]
indeed, but you'd have to write it like: 

```
fit_items = [
    {'name': "Values[k_on]", 'lower': 1e05, 'upper': 1e09, 'affected': ['Exp_1', 'Exp_2', 'Exp_3']},
    {'name': "Values[k_on]", 'lower': 1e05, 'upper': 1e09, 'affected': ['Exp_4']},
    {'name': "Values[k_on]", 'lower': 1e05, 'upper': 1e09, 'affected': ['Exp_5']},
]
```

**Comment by Theo-BRN** [comment]
Oh of course, tha's just a silly mistake of mine. 

In which case that's brilliant functionality! Thank you so much for your help!

Are you able to supply a pandas dataframe to the function instead of a list of dictionaries?

**Comment by fbergmann** [comment]
oh i missed this question. Indeed the fititems can be set as pandas dataframe
